openlibrary/templates/account/reading_log.html

```diff
@@ -71,7 +71,21 @@
       $:macros.Pager(int(current_page), doc_count, results_per_page=results_per_page)
   $else:
     <ul class="list-books">
-      $if checkin_year:
+      $if q:
+        <form method="GET" class="olform pagesearchbox">
+          <input type="text" minlength="3" placeholder="$_('Search your reading log')" name="q" value="$(query_param('q', ''))"/>
+          <input type="submit"/>
+        </form>
+        $ query = query_param('q')
+        <center>
+          <hr>
+          <div class="red">$_("No results found in %s" %key) </div>
+          <hr>
+          <div>
+            <a href="/search/inside?$urlencode(dict(q=query))">$_('Search all of Open Library for "%s"?' % query)</a>
```

Comment:
```suggestion
            <a href="/search?$urlencode(dict(q=query))">$_('Search all of Open Library for "%s"?' % query)</a>
```
Use book search instead of full-text search.